addons/mail/static/src/messaging/entity/follower/follower.js

```diff
@@ -0,0 +1,175 @@
+odoo.define('mail.messaging.entity.Follower', function (require) {
+'use strict';
+
+const { registerNewEntity } = require('mail.messaging.entity.core');
+
+function FollowerFactory({ Entity }) {
+
+    class Follower extends Entity {
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @return {string}
+         */
+        get name() {
+            if (this.channel) {
+                return this.channel.name;
+            }
+            return this.partner.name;
+        }
+
+        async remove() {
+            await this.env.rpc({
+                model: this.thread.model,
+                method: 'message_unsubscribe',
+                args: [
+                    [this.thread.id],
+                    this.partner ? [this.partner.id] : [],
+                    this.channel ? [this.channel.id] : [],
+                ],
+            });
+            this.delete();
+        }
+
+        async showSubtypes() {
+            const subtypesData = await this.env.rpc({
+                route: '/mail/read_subscription_data',
+                params: { follower_id: this.id },
+            });
+            this.unlink({ subtypes: null });
+            for (const data of subtypesData) {
+                const subtype = this.env.entities.FollowerSubtype.insert(data);
+                this.link({ subtypes: subtype });
+                if (data.followed) {
+                    this.link({ checkedSubtypes: subtype });
+                } else {
+                    this.unlink({ checkedSubtypes: subtype });
+                }
+            }
+            this.env.entities.Dialog.open('FollowerSubtypesEditor', { follower: this });
+        }
+
+        /**
+         * @param {mail.messaging.entity.FollowerSubtype} subtype
+         */
+        toggleSubtypeCheck(subtype) {
+            if (this.checkedSubtypes.includes(subtype)) {
+                this.unlink({ checkedSubtypes: subtype });
+            } else {
+                this.link({ checkedSubtypes: subtype });
+            }
+        }
+
+        async updateFollowerSubtypes() {
+            if (!this.checkedSubtypes) {
+                this.unlink({ thread: null });
+            }
+            const kwargs = {
+                subtype_ids: this.checkedSubtypes.map(subtype => subtype.id)
+            };
+            if (this.partner) {
+                kwargs.partner_ids = [this.partner.id];
+            } else {
+                kwargs.channel_ids = [this.channel.id];
+            }
+            await this.env.rpc({
```

Comment:
I think we should start commenting when the result of an RPC is coming from the bus (is is the case here)? Because I often wonder if we forgot to update stuff after or not.

Reply:
no update from longpolling in this case, but agree it's hard to tell whether there is a side-effect or not

Reply:
Didn't understand exactly... Seeing that it's outdated, is there still something to do ?

Reply:
Basically: the server is the source of truth. Therefore it is strange to call an RPC that supposedly leads to state change, but nothing resulting from it is updated.

Sometimes this is not the case because the result is coming from the longpolling as opposed to being a result of the RPC. I would like that we start to comment the RPC call when this is the case, to explain case by case where the result is handled.

And it appears that here is yet another case: we just update the store before hand, and hope that the RPC works, without even checking?
That seems like a bad idea.

Reply:
Yeah, ideally JS should handle toggle of checkboxes as temporary measures, but it shouldn't alter relations until application of subtype changes.

Reply:
For what I see in the code now, seems that the result of the rpc is actually used